core/promise/promise.go

```diff
@@ -65,3 +74,41 @@ func (sp *SignedPromise) Send(sender communication.Sender) error {
 
 	return nil
 }
+
+// Validate check signed promise to be valid. It checks signature, benefiter address.
+// Also it compares the promised amount to be enough for the proposal.
+// And finally it checks that issuer have enough balance to issue the promice.
+func (sp *SignedPromise) Validate(proposal dto.ServiceProposal, balance identity.Balance) error {
```

Comment:
It feels like this validate does too much. Why it has to depend on proposal? Signed promise with valid signature itself is perfectly valid, as both parties agreed on amount and other parameters